cmd/ci-secret-generator/main.go

```diff
@@ -120,17 +120,36 @@ func (o *options) validateConfig() error {
 }
 
 func executeCommand(command string) ([]byte, error) {
-	out, err := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", command).CombinedOutput()
-	if err != nil {
-		return nil, fmt.Errorf("%s : %w", string(out), err)
+	cmd := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", command)
+	var outBuf, errBuf bytes.Buffer
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	if err := cmd.Run(); err != nil {
+		// The command completed with non zero exit code, standard streams *should* be available.
+		if _, ok := err.(*exec.ExitError); ok {
+			errStr := errBuf.String()
+			if len(errStr) == 0 {
+				return nil, fmt.Errorf(" : %w", err)
+			}
+			return nil, fmt.Errorf("%s: %w", errStr, err)
+		}
+		// At this point we can't easily tell why the command failed,
+		// there is no guarantee neither stdout nor stderr are valid.
+		return nil, fmt.Errorf("failed to run the command: %w", err)
```

Comment:
I know some of these already exist, but let's turn them into decent messages, since we're changing this code.

They should always include the command which was executed (or not, if the caller does, but we have both here).

Reply:
Yeah, that code sucks and it exists only to pass tests.

Reply:
Look fine, I'll do that.